docs/specifications/tests/DNSQueryAndResponsSpecification.md

```diff
@@ -0,0 +1,154 @@
+
+
+## Default setting in *DNSSEC Query*
+
+An *DNSSEC Query* inherit the default setting from an *EDNS Query* except for the
+setting items specified below.
+
+A test case specification can refer to an *DNSSEC Query* with one or several
+changes to the setting items. Specifically, the test case specification must
+always specify query name and query type.
+
+|Setting               |Default value |Comment                   |
+|:---------------------|:-------|:-------------------------------|
+|ENDS DO flag          | set    |                                |
+
+
+## Default handling of a *DNS Response*
+
+A *DNS Response* is a response to a *DNS Query*. Unless specified in the test
+case specification, the items in the response is handled as listed.
+
+|Response Item |Default handling    | Comment                         |
+|:-------------|:-------------------|:--------------------------------|
+|OpCode        | Must be "response" | Always "response" in a response |
+|QR flag       | Must be set        | Always set in a response        |
+|AA flag       | -                  | Defined in test case            |
+|TC flag       | Re-query over TCP   |                                 |
+|RD flag       | ignore             |                                 |
+|RA flag       | ignore             |                                 |
+|AD flag       | ignore             |                                 |
+|CD flag       | ignore             |                                 |
+|RCODE         | -                  | Defined in test case            |
+|Query name    | ignore             |                                 |
+|Query type    | ignore             |                                 |
+|ENDS          | ignore             |                                 |
+
+
+* Check against query name and query type is, by default, done against the values
+  in the query, not in the response.
+  
+* When fetching records from the answer section, these are the default criteria:
+  * Only records matching the query name and the query type are fetched. RRSIG
+    records, when available, meeting the next criterium are also fetched.
+  * RRSIG records with the same query name and covering the query type are
+    fetched.
+  * CNAME records are ignored unless the query type is CNAME.
+
+* When CNAME chains are followed (by specification) and fetch from the answer
+  section, these are the default criteria:
+  * The first CNAME in the chain must match the query name.
+  * The last record in the chain must either be a CNAME or a record matching the
+    query type.
+
+* Authority and additional sections are by default ignored.
+
+
+## Default handling of an *EDNS Response*
+
+A *EDNS response* is a response to an *EDNS Query*. An *EDNS Response* inherits
+the default handling from a *DNS Response* except for the response items
+specified below. Unless specified in the test case specification, the items in
+the response is handled as the default handling.
```

Comment:
`as` -> `using`

There's another one like this below.

Reply:
Fixed.